File: src/main/java/com/movisens/xs/api/exceptions/MovisensXSException.java
package com.movisens.xs.api.exceptions;

import com.movisens.xs.api.models.ApiError;

import java.io.IOException;

public class MovisensXSException extends IOException {

    private static final long serialVersionUID = 1359714572930063286L;

    public MovisensXSException(ApiError apiError) {
        super(parseMessage(apiError));

    }

    private static String parseMessage(ApiError apiError) {
        if (apiError != null) {
            return apiError.getMessage();
        } else {
            return new ApiError("unknown", "unknown").getMessage();
        }
    }

    public MovisensXSException() {
        super();
    }

}

File: src/main/java/com/movisens/xs/api/exceptions/MovisensXSException.kt
package com.movisens.xs.api.exceptions

import com.movisens.xs.api.models.ApiError
import com.movisens.xs.api.exceptions.MovisensXSException
import java.io.IOException

open class MovisensXSException : IOException {
    constructor(apiError: ApiError?) : super(parseMessage(apiError))
    constructor() : super()

    companion object {
        private const val serialVersionUID = 1359714572930063286L
        private fun parseMessage(apiError: ApiError?): String {
            return if (apiError != null) {
                apiError.message
            } else {
                ApiError("unknown", "unknown").message
            }
        }
    }
}